tests/test_s3.py

```diff
@@ -1543,6 +1543,164 @@ def test_has_statement_policy(self):
+        self.addCleanup(destroyBucket, client, replicated_to_name)
+
+        # Setup buckets for replication by creating and adding versioning
+        for bucket in (replicated_from_name, replicated_to_name):
+            client.create_bucket(
+                Bucket=bucket
+            )
+            client.put_bucket_versioning(
+                Bucket=bucket,
+                VersioningConfiguration={
+                    'MFADelete': 'Disabled',
+                    'Status': 'Enabled'
+                }
+            )
+
+        # add a replication policy between them
+        client.put_bucket_replication(
+            Bucket=replicated_from_name,
+            ReplicationConfiguration={
+                'Role': 'arn:aws:iam::' + account_id + ':role/s3_replicator',
+                'Rules': [
+                    {
+                        'ID': 'test12345',
+                        'Prefix': '',
+                        'Status': 'Enabled',
+                        'Destination': {
+                            'Bucket': 'arn:aws:s3:::' + replicated_to_name
+                        }
+                    }
+                ]
+            }
+        )
+
+        # Test to make sure that a replication policy was in fact created between the buckets
+        response = client.get_bucket_replication(Bucket=replicated_from_name)
+        for rule in response['ReplicationConfiguration']['Rules']:
+            self.assertEqual(rule['Status'], 'Enabled')
+
+        p = self.load_policy(
+            {
+                "name": "s3-has-replica-policy",
+                "resource": "s3",
+                "filters": [
+                    {
+                        "type": "value",
+                        "key": "Replication.ReplicationConfiguration.Rules[].Destination",
+                        "value": "present"
+                    },
+                    {
+                        "type": "value",
+                        "key": "Replication.ReplicationConfiguration.Rules[].Status",
+                        "value": "Enabled",
+                        "op": "contains"
+                    }
+                ],
+                "actions": [
+                    {
+                        "type": "set-bucket-replication",
+                        "state": "remove"
+                    }
+                ]
+            },
+            session_factory=session_factory,
+        )
+        resources = p.run()
+        self.assertEqual(len(resources), 1)
+
+        # Test to make sure that the replication policy removed from the buckets
+        self.assertRaises(ClientError, client.get_bucket_replication, Bucket=replicated_from_name)
+
+    def test_bucket_replication_policy_disable(self):
+        bname = "repela"
+        self.patch(s3.S3, "executor_factory", MainThreadExecutor)
+        self.patch(
+            s3.SetBucketReplicationConfig, "executor_factory", MainThreadExecutor
+        )
+        augments = list(s3.S3_AUGMENT_TABLE)
```

Comment:
augments should be patched to the exact augment of interest for the test, ie replication

Reply:
Resolved in 4b8291831aae806235731e76a1928dbfb915723a. Good call, have no idea why I was trying to add that haha. Makes sense to peel it down as lean as possible.